<?xml version="1.0" encoding="utf-8"?>

<launch>

  <!-- Load Parameter -->
  
  <param name="yolov5_path"       value="$(find yolov5_ros)/yolov5"/>
  <param name="use_cpu"           value="false" />

  <!-- Start rosbag file -->
  <node pkg="rosbag" type="play" name="rosbag_play" args="$(find yolov5_ros)/scripts/lanedetection_v1.bag"/>
  
  <!-- left camera node -->
  <node pkg="yolov5_ros" type="L_image_process.py" name="L_img_node" output="screen" >
    <param name="weight_path"       value="$(find yolov5_ros)/weights/yolov5s.pt"/>
    <param name="image_topic_l"       value="/vds_node_localhost_2212/image_raw" />
    <param name="pub_topic"         value="/yolov5/BoundingBoxes" />
    <param name="camera_frame"       value="camera_color_frame"/>
    <param name="conf"              value="0.3" />
  </node>

  <!-- right camera node -->
  <node pkg="yolov5_ros" type="R_image_process.py" name="R_img_node" output="screen" >
      <param name="weight_path"       value="$(find yolov5_ros)/weights/yolov5s.pt"/>
      <param name="image_topic_r"       value="/vds_node_localhost_2212/image_raw" />
      <param name="pub_topic"         value="/yolov5/BoundingBoxes" />
      <param name="camera_frame"       value="camera_color_frame"/>
      <param name="conf"              value="0.3" />
    </node>

  <!-- output receiver -->
  <node pkg="yolov5_ros" type="output_recv.py" name="output_receiver" output="screen" >
      <param name="recv_topic"         value="/yolov5/BoundingBoxes" />
    </node>
</launch>

